### PR TITLE
[upstream-update] driver: adopt to new Archive API

### PR DESCRIPTION
**Pull request by compnerd**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

The archiving API was changed in SVN r275361.  Adjust accordingly.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci Please test

**Comment by modocache** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC @rudkx @gottesmm

**Comment by gottesmm** [comment]
(From reading Linux, it looks like you missed a spot). Assigning to Mark.

**Comment by compnerd** [comment]
Hmm, from my look at it, its due to the llvm branch not being far enough along (SVN r275361+).  Maybe Im misunderstanding something?

**Comment by gottesmm** [comment]
Ah ok. I think the issue is Mark needs to do the llvm/clang part of the update first and then merge this in.

File: tools/driver/autolink_extract_main.cpp [hunk 1]
```diff
@@ -149,7 +150,7 @@ static bool extractLinkerFlags(const llvm::object::Binary *Bin,
         return true;
       }
     }
-    return false;
+    return Error ? true : false;
```
Comment (rudkx) [inline tools/driver/autolink_extract_main.cpp]: Error appears to have a conversion to bool, so I don't think the select operation is needed.

**Comment by rudkx** [comment]
What I'd like to do with this is merge it locally onto my machine so that I get a clean build before pushing the merge.

Sound good?

File: tools/driver/autolink_extract_main.cpp (hunk 1 above)
Comment (compnerd) [inline tools/driver/autolink_extract_main.cpp]: Ah, right, didn't think of that.

**Comment by compnerd** [comment]
SGTM

File: tools/driver/autolink_extract_main.cpp (hunk 1 above)
Comment (rudkx) [inline tools/driver/autolink_extract_main.cpp]: Do you want me to fix it, or do you want to push an updated commit?

Comment (compnerd) [inline tools/driver/autolink_extract_main.cpp]: Updated :-)

Comment (rudkx) [inline tools/driver/autolink_extract_main.cpp]: Actually it's marked explicit, so I'll just put bool(Error) into your change.

**Comment by rudkx** [comment]
I ended up cherry-picking your change and tweaking the return to bool(Error) to use the explicit conversion.

**Pull request closed without merging**